client/hotrod-client-new/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java

```diff
@@ -821,4 +856,227 @@ public CompletionStage<Void> updateBloomFilter() {
    public String toString() {
       return "RemoteCache " + name;
    }
+
+   @Override
+   public CompletionStage<CacheConfiguration> configuration() {
+      return CompletableFutures.completedNull();
+   }
+
+   @Override
+   public CompletionStage<V> get(K key, CacheOptions options) {
+      assertRemoteCacheManagerIsStarted();
+      HotRodOperation<V> op = new AdvancedHotRodOperation<>(operationsFactory.newGetOperation(key), options);
```

Comment:
I like the use of AdvancedHotRodOperation here :) Just shows how nice the flexibility is.

Reply:
This approach with the delegate helped SO MUCH in this PR.